### PR TITLE
[5.7] Send an event when the user's email is verified

### PR DESCRIPTION
**Pull request by ThomHurks**
Upon user email verification we may like to take certain immediate actions. For example, once we have verified that the user has a valid email from a certain company domain, we may assign a role automatically.

I tried grepping for existing email verification tests, but couldn't find any. I prefer creating tests, but I think completely testing the email verification feature is outside the scope of this PR. If I missed the tests, please point me to them.

### CODE REVIEW

File: src/Illuminate/Auth/MustVerifyEmail.php [hunk 1]
```diff
@@ -17,11 +17,11 @@ public function hasVerifiedEmail()
     /**
      * Mark the given user's email as verified.
      *
-     * @return void
+     * @return bool
      */
     public function markEmailAsVerified()
     {
-        $this->forceFill(['email_verified_at' => $this->freshTimestamp()])->save();
+        return $this->forceFill(['email_verified_at' => $this->freshTimestamp()])->save();
```
Comment (m1guelpf) [inline src/Illuminate/Auth/MustVerifyEmail.php]: I'd say this is breaking?

**Comment by ThomHurks** [comment]
Thanks a lot @taylorotwell !

**Comment by ThomHurks** [comment]
@taylorotwell Just noticed that the PHPDoc in the contract should be adjusted too. Created PR #25463.

**Comment by antonkomarev** [comment]
Why method signature has been changed?
Isn't it better to throw CustomException if action couldn't be done?

**Comment by ThomHurks** [comment]
@antonkomarev I decided to change the original behaviour of the code as little as possible, as throwing an exception would change the flow completely.

As far as I understand the code, if there's an Observer watching the User model it can cancel the save by returning false; I didn't want the Verified event to still be emitted in that case, which is why I check for the result of the save action.

It is true that besides not emitting the Verified event, no other action is taken when the save fails. This is in line with the original code, where the result of the save is also ignored. If that behaviour needs to change then that could be solved in a different PR if you have a good idea about what would be better.

**Comment by antonkomarev** [comment]
@ThomHurks I'd prefer exceptions to know exact reason why this was happened. And if we just return boolean - it's a black box.

**Pull request merged**